src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -172,4 +176,16 @@ public String getOptionalArrayTypeName() {
   public String getDynamicReturnTypeName(Method method, MethodConfig methodConfig) {
     return getNotImplementedString("SurfaceNamer.getDynamicReturnTypeName");
   }
+
+  public String getApiSettingsClassName(Interface service) {
+    return className(Name.upperCamel(service.getSimpleName(), "Settings"));
+  }
+
+  public String getTestClassName(Interface service) {
+    return className(Name.upperCamel(service.getSimpleName(), "Test"));
+  }
+
+  public String getMockServiceClassName(Interface service) {
+    return className(Name.upperCamel("Mock", service.getSimpleName()));
+  }
```

Comment:
Ok, I'm convinced. Nevertheless, some thought should be put into documenting these methods. This class already has 30+ methods, and is not necessarily obvious to developers which is used for what without searching for usage and reading through the dependent code to determine the use cases.